### ISSUE TITLE
Airplay/Groups - Music plays on powered off speaker that's not part of  a group

### ISSUE THREAD

**Issue by brian-maher** [music-assistant/support#3244]
### What version of Music Assistant has the issue?

2.3.2

### What version of the Home Assistant Integration have you got installed?

2024.11

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When switching from an individual player to a group, music is streamed to the powered off players that were previously used, even if they aren't in the group. This behaviour wasn't present before the group re-working.

### How to reproduce

Preamble:

- 3 AirPlay Players: Player A, Player B, Player C
- 1 Group Player (Player B + C)

To Reproduce:

- Play music on Player A
- Stop Player A
- Power off Player A
- Play music on Group player (Player B + C).
- Music plays on Players A, B + C

Player A will show as "Powered Off" whilst playing.

### Music Providers

Spotify

### Player Providers

Airplay

### What version of Home Assistant Core are your running

2024.11

### What type of installation are you running?

Home Assistant Container

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
Sounds like somehow that other player was the group leader then. This will be looked at soon.

**Comment by marcelveldt** [comment]
Can this be retested in 2.4.0(rc5) ?

**Comment by OzGav** [comment]
@brian-maher Please advise if you can test with 2.4.0rc5 or later.

**Comment by OzGav** [comment]
@brian-maher we will have to close this soon if there is no follow up

**Comment by brian-maher** [comment]
I will try to test this tonight :)

**Comment by brian-maher** [comment]
@OzGav Had a chance to play with this at lunch. It's nearly there - but not sure if you want a new issue or not.

This appears to be fixed in Music Assistant itself - however it still occurs using the Home Assistant add-on (e.g. using `music_assistant.play_media`). The alternative, that should work I guess, is to un-join the player before starting music - but that powers off the group and `music_assistant.play_media` doesn't seem to power on group members when playing to a group that's powered off.

For reference, what I'm trying to do is something like:

- `music_assistant.play_media` target: single speaker
- `media_player.clear_playlist`
- `music_assistant.play_media` target: group

Not sure if this should be a separate issue or not :)

**Comment by OzGav** [comment]
So are you saying the single speaker is not part of the group but it is the same provider type and it starts playing? If so what does the player list show, does it show the extra speaker suddenly grouped or not at all?

Is everything as described in the original post but only happening with the actions now?

**Comment by marcelveldt** [comment]
If a player is synced, any playback related commands you send will always be redirected to the sync leader (in case of manually synced/grouped players).

If a player is part of one or more groups, there can only be one group active (=powered) at a time.
And if a group is active, all playback related commands to a group member player will be redirected to the group.

If all grouped are not powered and a player is not synced, if you issue a playback command, it should be handled by the player itself. Is that is still somehow handled by a group, that is a (very strange) bug.

**Comment by OzGav** [comment]
@brian-maher does Marcel’s response above explain what you are seeing?

**Comment by OzGav** [comment]
This will be closed soon assuming fixed

**Comment by brian-maher** [comment]
Apologies for the delay getting back - things have been mad so I haven't had a chance to get to this!

Thanks @marcelveldt  @OzGav  - I think someone has actually opened #3725 which I think explains my current issue a bit better so best this stays closed! I think it's basically a problem with the HA Addon not handling the join/unjoin operations properly - as everything works absolutely fine from MASS itself :) 

Thanks again for all your help!
